src/main/groovy/de/dkfz/eilslabs/batcheuphoria/execution/RestExecutionService.groovy

```diff
@@ -0,0 +1,207 @@
+import org.apache.http.impl.client.HttpClientBuilder
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager
+import org.apache.http.message.BasicHeader
+import org.apache.http.protocol.HTTP
+import org.apache.http.ssl.SSLContexts
+import org.apache.http.util.EntityUtils
+
+import javax.net.ssl.SSLContext
+import java.security.KeyStore
+
+/**
+ * Created by kaercher on 12.01.17.
+ */
+class RestExecutionService implements ExecutionService{
+
+    private static final LoggerWrapper logger = LoggerWrapper.getLogger(RestExecutionService.class.name);
+
+    public boolean isCertAuth = false
+
+    private String token = ""
+
+    private File keyStoreLocation
+    private String keyStorePassword
+
+    public final String BASE_URL
+
+    /*REST RESOURCES*/
+    public static final String RESOURCE_LOGON = "/logon"
+    public static final String RESOURCE_LOGOUT = "/logout"
+
+    RestExecutionService(String baseURL, String username, String password) throws AuthenticationException{
+        this.BASE_URL = baseURL
+        RestResult result = logon(username,password)
+        if(result.statusCode == 200){
+            this.token = new XmlSlurper().parseText(result.body).token.text()
+        }else{
+            logger.warning("Could not authenticate, returned HTTP status code: ${result?.statusCode}")
+            throw new AuthenticationException("Could not authenticate, returned HTTP status code: ${result?.statusCode}")
+        }
+    }
+
+
+    RestExecutionService(File keystoreLocation, String keyStorePassword){
+        this.keyStoreLocation = keystoreLocation
+        this.keyStorePassword = keyStorePassword
+
+    }
+
+    RestResult logon(String username, String password){
+        String url = RESOURCE_LOGON
+        String body = "<User><name>${username}</name><pass>${password}</pass></User>"
+        List<Header> headers = []
+        headers.add(new BasicHeader(HTTP.CONTENT_TYPE, "application/xml;charset=UTF-8"))
+        headers.add(new BasicHeader("Accept", "application/xml"))
+        return execute(new RestCommand(url,body,headers,RestCommand.HttpMethod.HTTPPOST))
+    }
+
+
+    boolean logout(){
+        String url = RESOURCE_LOGOUT
+        List<Header> headers = []
+        headers.add(new BasicHeader(HTTP.CONTENT_TYPE, "application/xml;charset=UTF-8"))
+        headers.add(new BasicHeader("Accept", "application/xml"))
+
+        RestResult result =execute(new RestCommand(url,null,headers,RestCommand.HttpMethod.HTTPPOST))
+        if(result.statusCode == 200){
+            this.token = null
+            return  true
+        }else{
+           return false
+        }
+    }
+
+
+    RestResult execute(RestCommand restCommand) {
+        String url = BASE_URL+restCommand.resource
+        println("url: "+url)
+        def httpClient
+        if(isCertAuth)
+            httpClient = createHttpClientWithClientCertAuth()
+        else
+            httpClient = HttpClientBuilder.create().build()
+
+        def httpRequest
```

Comment:
Yes, I know it's better to use defined types. But in this particular case it saved me some code. htttpClient can be HttpPost or HttpGet